app/models/team_zip_export.rb

```diff
@@ -0,0 +1,273 @@
+    name.gsub(/^-|\.+$| +$/, '_')
+  end
+
+  # Appends given suffix to file_name and then adds original extension
+  def append_suffix(file_name, suffix)
+    ext = File.extname(file_name)
+    File.basename(file_name, ext) + suffix + ext
+  end
+
+  # Helper method to extract given assets to the directory
+  def export_result_assets(assets, directory)
+    assets.each_with_index do |asset, i|
+      file = FileUtils.touch("#{directory}/#{append_suffix(asset.file_file_name,
+                                                           "_#{i}")}").first
+      File.open(file, 'wb') { |f| f.write(asset.open.read) }
+    end
+  end
+
+  # Helper method to extract given step assets to the directory
+  def export_step_assets(assets, directory)
+    assets.each_with_index do |step_asset, i|
+      asset = step_asset.asset
+      file = FileUtils.touch(
+        "#{directory}/" \
+        "#{append_suffix(asset.file_file_name,
+                         "_#{i}_Step#{step_asset.step.position + 1}")}"
+      ).first
+      File.open(file, 'wb') { |f| f.write(asset.open.read) }
+    end
+  end
+
+  # Helper method to extract given tables to the directory
+  def export_step_tables(step_tables, directory)
+    step_tables.each_with_index do |step_table, i|
+      table = step_table.table
+      table_name = table.name.presence || 'Table'
+      table_name += i.to_s
+      file = FileUtils.touch(
+        "#{directory}/#{handle_name(table_name)}" \
+        "_#{i}_Step#{step_table.step.position + 1}.csv"
+      ).first
+      File.open(file, 'wb') { |f| f.write(table.to_csv) }
+    end
+  end
+
+  # Helper method to extract given tables to the directory
+  def export_result_tables(tables, directory)
+    tables.each_with_index do |table, i|
+      table_name = table.name.presence || 'Table'
+      table_name += i.to_s
+      file = FileUtils.touch("#{directory}/#{handle_name(table_name)}.csv")
+                      .first
+      File.open(file, 'wb') { |f| f.write(table.to_csv) }
+    end
+  end
+
+  # Helper method for saving inventories to CSV
+  def save_inventories_to_csv(path, repo, repo_rows, id)
+    repo_name = handle_name(repo.name) + "_#{id}"
+    file = FileUtils.touch("#{path}/#{repo_name}.csv").first
+
+    # Attachment folder
+    rel_attach_path = "#{repo_name}_ATTACHMENTS"
+    attach_path = "#{path}/#{rel_attach_path}"
+    FileUtils.mkdir_p(attach_path)
+
+    # Define headers and columns IDs
+    col_ids = [-3, -4, -5, -6] + repo.repository_columns.map(&:id)
+
+    # Define callback function for file name
+    assets = {}
+    asset_counter = 0
+    handle_name_func = lambda do |asset|
+      file_name = append_suffix(asset.file_file_name, "_#{asset_counter}").to_s
+
+      # Save pair for downloading it later
+      assets[asset] = "#{attach_path}/#{file_name}"
+
+      asset_counter += 1
+      rel_path = "#{rel_attach_path}/#{file_name}"
+      return "=HYPERLINK(\"#{rel_path}\", \"#{rel_path}\")"
```

Comment:
Does this actually open a file in Excel? Because it doesn't with `LibreOffice Calc`.